message.go

```diff
@@ -18,6 +18,29 @@ const messageStructSize = int(unsafe.Sizeof(RedisMessage{}))
 // Nil represents a Redis Nil message
 var Nil = &RedisError{typ: typeNull}
 
+// default values for Errors
+const (
+	defaultErrorBool    = false
+	defaultErrorString  = "ERR"
+	defaultErrorInt64   = int64(0)
+	defaultErrorFloat64 = float64(0.0)
+)
+
+var defaultErrorRedisMessageSlice []RedisMessage
+var defaultErrorXRangeEntrySliceMap = make(map[string][]XRangeEntry)
+var defaultErrorZScoreSliceMap = make(map[string][]ZScore)
+var defaultErrorZScore = ZScore{}
+var defaultErrorRedisMessageMap = make(map[string]RedisMessage)
+var defaultErrorScanEntry = ScanEntry{}
+var defaultErrorStrMap = make(map[string]string)
+var defaultErrorStrMapSlice = make([]map[string]string, 0)
+var defaultErrorIntMap = make(map[string]int64)
+var defaultErrorKeyValues = KeyValues{}
+var defaultErrorKeyZScores = KeyZScores{}
```

Comment:
On retrospect this might not be a great idea at all

Maybe I misunderstood the proposal a little

Reply:
We don't need to pre-define these default error values. We should just return a zero value of the corresponding type as well as the error. For example:

`return defaultErrorRedisMessageSlice, fmt.Errorf("redis message type %s is not a array", typeNames[typ])`

should be just

`return nil, fmt.Errorf("redis message type %s is not a array", typeNames[typ])`

Reply:
@rueian Zero values for the non-nullable ones? 

It will not compile in those cases when we send back a nil for non-nullable one I guess
Okay let me do try this refactor

Reply:
Also, is this the only file where we require changes for this? @rueian 

Any other changes we require other than the tests

Reply:
Yes, the `message.go` is the only file that needs to be changed other than the tests.